File: JAVA/Test/Graph/UnionFindTest.java
package Test.Graph;

import Source.IO.MyScanner;
import Source.Graph.UnionFind;

import java.util.*;

public class UnionFindTest
{
	public static void main(String[] args)
	{
		MyScanner sc = new MyScanner();
		int t = sc.nextInt();
		for(int j=1;j<=t;j++)
		{
			int n = sc.nextInt();
			int m = sc.nextInt();

			UnionFind uf = new UnionFind(n); 
			for(int i=0;i<m;i++)
			{
				int a = sc.nextInt();
				int b = sc.nextInt();
				a--;b--;
				uf.join(a,b);
			}

			ArrayList<ArrayList<Integer>> components = uf.getComponents();
			for(ArrayList<Integer> component:components)
			{
				for(Integer ver:component)
				{
					System.out.print((ver+1)+" ");
				}
				System.out.println();
			}
		}
		sc.close();
	}
}

File: C++/Test/Graph/UnionFindTest.cpp
#include<iostream>
#include<vector>
#include "../../Source/IO/MyScanner.h"
#include "../../Source/Graph/UnionFind.h"

using namespace std;

int main()
{
	MyScanner sc;
	int t = sc.nextInt();
	while(t--)
	{
		int n = sc.nextInt();
		int m = sc.nextInt();

		UnionFind uf(n); 
		for(int i=0;i<m;i++)
		{
			int a = sc.nextInt();
			int b = sc.nextInt();
			a--;b--;
			uf.join(a,b);
		}

		vector<vector<int> > components = uf.getComponents();
		for(int i=0;i<components.size();i++)
		{
			for(int j=0;j<components[i].size();j++)
			{
				cout<<(components[i][j]+1)<<" ";
			}
			cout<<endl;
		}
	}
}